From 1fce50c4f04ba93384c2f3bb4b4814f1fe328e5b Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Sun, 5 Oct 2014 20:35:23 -0700
Subject: [PATCH] Add support to pass old and new bounds in WindowDelegate.

OnBoundSChanged expects newbounds of the window. However, we need old
and new bounds in DRWH to decide if we need to resize, move or just update.
This patch adds a new api in PlatformWindowDelegate to do this for now.
Once, the migration to PlatformWindow is completed, lets se if we can
upstream this.
---
 ui/platform_window/platform_window_delegate.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ui/platform_window/platform_window_delegate.h b/ui/platform_window/platform_window_delegate.h
index b215275..3429b7c 100644
--- a/ui/platform_window/platform_window_delegate.h
+++ b/ui/platform_window/platform_window_delegate.h
@@ -29,6 +29,8 @@ class PlatformWindowDelegate {

   // Note that |new_bounds| is in physical screen coordinates.
   virtual void OnBoundsChanged(const gfx::Rect& new_bounds) = 0;
+  virtual void OnBoundChanged(const gfx::Rect& old_bounds,
+                              const gfx::Rect& new_bounds) { }

   // Note that |damaged_region| is in the platform-window's coordinates, in
   // physical pixels.
--
1.9.1
